[CELEBORN-1425][HELM] Add helm chart unit tests to ensure manifests are rendered as expected  Edit  Add comment (#2499)
apache/celeborn: charts/celeborn/README.md

under the License.
-->

# celeborn

![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.4.0](https://img.shields.io/badge/AppVersion-0.4.0-informational?style=flat-square)

> Celeborn is dedicated to improving the efficiency and elasticity of different map-reduce engines. Celeborn provides an elastic and high efficient management service for shuffle data.

Reviewer: Better keep this, and the content here should be consistent with the content on the Celeborn website see [Apache Celeborn](https://celeborn.apache.org/)

Author: > Better keep this, and the content here should be consistent with the content on the Celeborn website see [Apache Celeborn](https://celeborn.apache.org/)

Updated.

under the License.
-->

> # celeborn

Reviewer 2: Revert to `Helm Chart for Apache Celeborn`.

![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.4.0](https://img.shields.io/badge/AppVersion-0.4.0-informational?style=flat-square)

> Celeborn is an intermediate data service for Big Data compute engines (i.e. ETL, OLAP and Streaming engines) to boost performance, stability, and flexibility. Intermediate data typically include shuffle and spilled data.

Reviewer 2: Line 24-26 revert to `[Apache Celeborn](https://celeborn.apache.org) is an intermediate data service for Big Data compute engines (i.e. ETL, OLAP and Streaming engines) to boost performance, stability, and flexibility. Intermediate data typically include shuffle and spilled data.`.

The chart can be customized using the following [celeborn configurations](https://celeborn.apache.org/docs/latest/configuration/#important-configurations).
Specify parameters using `--set key=value[,key=value]` argument to `helm install`.

## Documentation

For additional details on deploying the Celeborn Kubernetes Helm chart, please refer to the [Celeborn on Kubernetes](https://celeborn.apache.org/docs/latest/deploy_on_k8s/) documentation.

## Values

| Key | Type | Default | Description |
|-----|------|---------|-------------|
| affinity | object | `{}` | Pod affinity |
| cluster.name | string | `"cluster"` |  |
| dnsPolicy | string | `"ClusterFirst"` | Specifies the DNS policy for Celeborn pods to use |
| fullnameOverride | string | `""` | String to override the default generated fullname |
| hostNetwork | bool | `false` | Specifies whether to use the host's network namespace |
| image.pullPolicy | string | `"Always"` | Image pull policy |
| image.repository | string | `"aliyunemr/remote-shuffle-service"` | Image repository |
| image.tag | string | `"0.1.1-6badd20"` | Image tag |
| imagePullSecrets | list | `[]` | Image pull secrets for private image registry |

> | masterReplicas | int | `3` | Specifies the number of Celeborn master replicas to deploy, master replicas should not less than 3 |

Reviewer: Why the key is still masterReplicas?

Author: > Why the key is still masterReplicas?

Sorry, I forget to run [helm-docs](https://github.com/norwoodj/helm-docs) go generate the latest documentation. And a pre-commit git hook can be configured to run this command automatically before commit.

Reviewer 3: what if the developer does not have [helm-docs](https://github.com/norwoodj/helm-docs) installed locally?

| resources.master | string | `nil` | Celeborn master pod resources |
| resources.worker | object | `{}` | Celeborn worker pod resources |
| securityContext.fsGroup | int | `10006` | Specifies the group ID to use when modifying ownership and permissions of the mounted volumes |
| securityContext.runAsGroup | int | `10006` | Specifies the group ID to run the entrypoint of the container process |
| securityContext.runAsUser | int | `10006` | Specifies the user ID to run the entrypoint of the container process |
| service.port | int | `9097` | Specifies service port |
| service.type | string | `"ClusterIP"` | Specifies service type |
| tolerations | list | `[]` | Pod tolerations |
| volumes | object | `{}` | Specifies Celeborn volumes. Current Celeborn support followings volume types are `emptyDir` and `hostPath`. <br> Note that `hostPath` only works in hostPath type using to set `volumes hostPath path`. Celeborn Master will pick first volumes for store raft log. <br> `diskType` only works in Celeborn Worker with hostPath type to manifest local disk type |

> | workerReplicas | int | `5` | Specifies the number of Celeborn worker replicas to deploy, should less than node number |

Reviewer: ditto.

## Documentation

For additional details on deploying the Celeborn Kubernetes Helm chart, please refer to the [Celeborn on Kubernetes](https://celeborn.apache.org/docs/latest/deploy_on_k8s/) documentation.

## Values

| Key | Type | Default | Description |
|-----|------|---------|-------------|
| affinity | object | `{}` | Pod affinity |
| cluster.name | string | `"cluster"` | Specifies Kubernetes cluster name |
| dnsPolicy | string | `"ClusterFirst"` | Specifies the DNS policy for Celeborn pods to use |
| fullnameOverride | string | `""` | String to override the default generated fullname |
| hostNetwork | bool | `false` | Specifies whether to use the host's network namespace |
| image.pullPolicy | string | `"Always"` | Image pull policy |
| image.pullSecrets | list | `[]` | Image pull secrets for private image registry |
| image.repository | string | `"aliyunemr/remote-shuffle-service"` | Image repository |
| image.tag | string | `"0.1.1-6badd20"` | Image tag |
| nameOverride | string | `""` | String to override the default generated name |
| nodeSelector | object | `{}` | Pod node selector |
| podAnnotations | object | `{}` | Pod annotations |

> | podMonitor.enable | bool | `true` | Specifies whether a PodMonitor should be created |

Reviewer 2: ```suggestion
| podMonitor.enable | bool | `true` | Specifies whether to enable creating a PodMonitor |
```

| podMonitor.podMetricsEndpoint | object | `{"interval":"5s","portName":"metrics","scheme":"http"}` | Specifies PodMonitor endpoint |
| priorityClass.master.create | bool | `false` | Specifies whether a new priority class for Celeborn master pods should be created |
| priorityClass.master.name | string | `""` | Specifies the name of priority class for Celeborn master pods to be used (created if `create: true`) |

> | priorityClass.master.value | int | `1000000000` | Specifies the integer value of this priority class, default is half of system-cluster-critical |

Reviewer 2: Could the description mention which values the priority class has?

Author: > Could the description mention which values the priority class has?

Added `empty means "${Release.Name}-master-priority-class"` in description.

| priorityClass.worker.name | string | `""` | Specifies the name of priority class for Celeborn worker pods to be used (created if `create: true`) |
| priorityClass.worker.value | int | `999999000` | Specifies the integer value of this priority class, default is Celeborn master value minus 1000 |
| replicaCount.master | int | `3` | Specifies the number of Celeborn master replicas to deploy, master replicas should not less than 3 |
| replicaCount.worker | int | `5` | Specifies the number of Celeborn worker replicas to deploy, should less than node number |
| resources.master | object | `{}` | Celeborn master pod resources |
| resources.worker | object | `{}` | Celeborn worker pod resources |
| securityContext.fsGroup | int | `10006` | Specifies the group ID to use when modifying ownership and permissions of the mounted volumes |
| securityContext.runAsGroup | int | `10006` | Specifies the group ID to run the entrypoint of the container process |
| securityContext.runAsUser | int | `10006` | Specifies the user ID to run the entrypoint of the container process |

> | service.port | int | `9097` | Specifies service port |

Reviewer 2: Could this port distingish the port of the Master and Worker service?

Author: > Could this port distingish the port of the Master and Worker service?

Actually, `service.port` is used only by the master service, the worker service does not expose a service port.

Reviewer 2: @ChenYi015, could the worker service expose a service port?

Author: @SteNicholas Just wondering, why do we need to expose a port for the worker service?

Reviewer 2: @ChenYi015, could the users deploy worker service in other k8s cluster?

Author: @SteNicholas Currently not, we need to modify the helm chart so that we can deploy celeborn workers in another k8s cluster.

Reviewer 2: @ChenYi015, would you like to support this in another pull request?

Author: @SteNicholas If that's you need, I will try to make it happen.
